Tests/Integration/JoinTests.cs

```diff
@@ -193,10 +190,8 @@ public async Task Join_And_Send_Unconfirmed_And_Confirmed_Messages(string device
             var downstreamMessage = PacketForwarder.DownlinkMessages[1];
 
             // validates txpk according to eu region
-#pragma warning disable CS0618 // #655 - This Rxpk based implementation will go away as soon as the complete LNS implementation is done
-            Assert.True(RegionManager.EU868.TryGetDownstreamChannelFrequency(confirmedMessageRxpk, out var frequency));
-#pragma warning restore CS0618 // #655 - This Rxpk based implementation will go away as soon as the complete LNS implementation is done
-            Assert.Equal(frequency, downstreamMessage.Txpk.Freq);
+            Assert.True(RegionManager.EU868.TryGetDownstreamChannelFrequency(radiometata.Frequency, out var frequency));
+            Assert.Equal(frequency, Hertz.Mega(downstreamMessage.Txpk.Freq));
```

Comment:
You can use `Txpk.FreqHertz` directly:

```suggestion
            Assert.Equal(frequency, downstreamMessage.Txpk.FreqHertz);
```